### ISSUE TITLE
Segfault while running the Tribler unit tests on current Debian SID AMD64

### ISSUE THREAD

**Issue by whirm** [libswift/libswift#37]
```
└──> gdb swift /tmp/core 
GNU gdb (Debian 7.7.1-2) 7.7.1
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from swift...(no debugging symbols found)...done.
[New LWP 22115]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `/var/data/feina/code/delft/tribler/swift -j -l 0.0.0.0:8962 -c 127.0.0.1:40473'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  memchr () at ../sysdeps/x86_64/memchr.S:154
154 ../sysdeps/x86_64/memchr.S: No such file or directory.
(gdb) bt
#0  memchr () at ../sysdeps/x86_64/memchr.S:154
#1  0x00007f3d33d0fdff in evbuffer_search_range (buffer=buffer@entry=0x6de740, what=what@entry=0x7f3d33d30de5 "\r\n", len=len@entry=0x2, 
    start=start@entry=0x7fffe8e96890, end=end@entry=0x0) at buffer.c:2527
#2  0x00007f3d33d0ff4c in evbuffer_search (buffer=buffer@entry=0x6de740, what=what@entry=0x7f3d33d30de5 "\r\n", len=len@entry=0x2, 
    start=start@entry=0x7fffe8e96890) at buffer.c:2493
#3  0x00007f3d33d1028e in evbuffer_search_eol (buffer=buffer@entry=0x6de740, start=start@entry=0x0, eol_len_out=eol_len_out@entry=0x7fffe8e968f8, 
    eol_style=eol_style@entry=EVBUFFER_EOL_CRLF_STRICT) at buffer.c:1444
#4  0x00007f3d33d1032f in evbuffer_readln (buffer=0x6de740, n_read_out=0x7fffe8e96960, eol_style=EVBUFFER_EOL_CRLF_STRICT) at buffer.c:1505
#5  0x000000000042cfae in CmdGwReadLine(int) ()
#6  0x000000000042cee2 in CmdGwProcessData(int) ()
#7  0x000000000042cebc in CmdGwDataCameInCallback(bufferevent*, void*) ()
#8  0x00007f3d33d12b6a in bufferevent_readcb (fd=0x14, event=<optimized out>, arg=0x6de890) at bufferevent_sock.c:186
#9  0x00007f3d33d08184 in event_process_active_single_queue (activeq=0x6ddfb0, base=0x6ddb70) at event.c:1350
#10 event_process_active (base=<optimized out>) at event.c:1420
#11 event_base_loop (base=0x6ddb70, flags=0x0) at event.c:1621
#12 0x0000000000409131 in utf8main(int, char**) ()
#13 0x000000000040c8c6 in main ()
(gdb) 
```

**Comment by whirm** [comment]
Another one:

```
(gdb) bt
#0  0x0000000000000000 in ?? ()
#1  0x00007f3cfe8f630e in evbuffer_readln (buffer=0x13cf740, n_read_out=0x7fff8f94d330, eol_style=EVBUFFER_EOL_CRLF_STRICT) at buffer.c:1499
#2  0x000000000042cfae in CmdGwReadLine(int) ()
#3  0x000000000042cee2 in CmdGwProcessData(int) ()
#4  0x000000000042cebc in CmdGwDataCameInCallback(bufferevent*, void*) ()
#5  0x00007f3cfe8f8b6a in bufferevent_readcb (fd=0x15, event=<optimized out>, arg=0x13cf890) at bufferevent_sock.c:186
#6  0x00007f3cfe8ee184 in event_process_active_single_queue (activeq=0x13cefb0, base=0x13ceb70) at event.c:1350
#7  event_process_active (base=<optimized out>) at event.c:1420
#8  event_base_loop (base=0x13ceb70, flags=0x0) at event.c:1621
#9  0x0000000000409131 in utf8main(int, char**) ()
#10 0x000000000040c8c6 in main ()
(gdb) 
```

**Comment by whirm** [comment]
closed by  #38
